<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>20&nbsp; Generalized linear model – R for Medical Statistics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./survival-analysis.html" rel="next">
<link href="./logistic-regression.html" rel="prev">
<link href="./favicon.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-29e2c20b02301cfff04dc8050bf30c7e.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-43df0d6f53b733b42d37fa7808e17715.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-218caa77794a2cebf88bf41099f21b16.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-612253c927da9ed618098f132c689ac9.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.css">

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./advanced-stat.html">Advanced methods</a></li><li class="breadcrumb-item"><a href="./generalized-linear-model.html"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Generalized linear model</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">R for Medical Statistics</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://qbgaoo.github.io/r4ms" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-git"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preface.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Preface</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./basic-stat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Basic methods</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./quat-data-statdesc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Statistical description of quantitative data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./normal-distribution.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Normal distribution</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./statistical-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Statistical inference</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./hypothesis-test.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Hypothesis test</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ttest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">t-test</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Analysis of Variance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./count-data-statdesc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Statistical description of count data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discrete-data-dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Distributions of discrete variables</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chi-square-test.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Chi-square test</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nonparametric-test.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Nonparametric Tests</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./regression-correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Bivariate regression and correlation</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./advanced-stat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Advanced methods</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multifactor-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Multifactor ANOVA</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./repeated-measurement.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Repeated measures ANOVA</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general-linear-model.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">General linear model</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multivariate-groups-diff.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Multivariate data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multiple-linear-regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Multiple linear regression analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./logistic-regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Logistic regression analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./generalized-linear-model.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Generalized linear model</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./survival-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Survival analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cluster-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Cluster analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discriminant-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Discriminant analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pca-factor-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Principal components and factor analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./canonical-correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Canonical correlation analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./structural-equation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Structural equation model</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multilevel-model.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Multilevel model</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./comprehensive-evaluation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Comprehensive evaluation methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./community-intervention.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Community intervention trials</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scale-develop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Medical scales</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./meta-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">31</span>&nbsp; <span class="chapter-title">Meta analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./general-estimat-equation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">32</span>&nbsp; <span class="chapter-title">Generalized estimating equation</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./practical-skills.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Practical skills</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-preprocess.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">33</span>&nbsp; <span class="chapter-title">Data preprocessing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./random-sampling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">34</span>&nbsp; <span class="chapter-title">Random sampling</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./randomization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">35</span>&nbsp; <span class="chapter-title">Randomization</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./sample-size.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">36</span>&nbsp; <span class="chapter-title">Sample size estimation</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pilot-project.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">37</span>&nbsp; <span class="chapter-title">A case to analyze</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./quarto.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">38</span>&nbsp; <span class="chapter-title">Quarto</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./memo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">39</span>&nbsp; <span class="chapter-title">Memo for future programming</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#prerequisite" id="toc-prerequisite" class="nav-link active" data-scroll-target="#prerequisite"><span class="header-section-number">20.1</span> Prerequisite</a></li>
  <li><a href="#components-of-glm" id="toc-components-of-glm" class="nav-link" data-scroll-target="#components-of-glm"><span class="header-section-number">20.2</span> Components of GLM</a></li>
  <li><a href="#log-linear-model" id="toc-log-linear-model" class="nav-link" data-scroll-target="#log-linear-model"><span class="header-section-number">20.3</span> Log-linear model</a>
  <ul class="collapse">
  <li><a href="#goodness-of-fit-test" id="toc-goodness-of-fit-test" class="nav-link" data-scroll-target="#goodness-of-fit-test"><span class="header-section-number">20.3.1</span> Goodness-of-fit test</a></li>
  <li><a href="#parameter-estimation" id="toc-parameter-estimation" class="nav-link" data-scroll-target="#parameter-estimation"><span class="header-section-number">20.3.2</span> Parameter estimation</a></li>
  </ul></li>
  <li><a href="#poisson-regression" id="toc-poisson-regression" class="nav-link" data-scroll-target="#poisson-regression"><span class="header-section-number">20.4</span> Poisson regression</a>
  <ul class="collapse">
  <li><a href="#goodness-of-fit-test-1" id="toc-goodness-of-fit-test-1" class="nav-link" data-scroll-target="#goodness-of-fit-test-1"><span class="header-section-number">20.4.1</span> Goodness-of-fit test</a></li>
  <li><a href="#parameters-estimation" id="toc-parameters-estimation" class="nav-link" data-scroll-target="#parameters-estimation"><span class="header-section-number">20.4.2</span> Parameters estimation</a></li>
  </ul></li>
  <li><a href="#negative-binomial-regression" id="toc-negative-binomial-regression" class="nav-link" data-scroll-target="#negative-binomial-regression"><span class="header-section-number">20.5</span> Negative binomial regression</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://qbgaoo.github.io/r4ms/edit/main/generalized-linear-model.qmd" class="toc-action"><i class="bi bi-git"></i>Edit this page</a></li><li><a href="https://qbgaoo.github.io/r4ms/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./advanced-stat.html">Advanced methods</a></li><li class="breadcrumb-item"><a href="./generalized-linear-model.html"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Generalized linear model</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Generalized linear model</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>A generalized linear model (GLM) is a flexible generalization of ordinary linear regression that allows for response variables that have error distribution models other than a normal distribution. GLMs are widely used in various fields, including medical statistics, economics, and social sciences, as they provide a robust framework to model different types of data and relationships.</p>
<section id="prerequisite" class="level2" data-number="20.1">
<h2 data-number="20.1" class="anchored" data-anchor-id="prerequisite"><span class="header-section-number">20.1</span> Prerequisite</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="components-of-glm" class="level2" data-number="20.2">
<h2 data-number="20.2" class="anchored" data-anchor-id="components-of-glm"><span class="header-section-number">20.2</span> Components of GLM</h2>
<p>A generalized linear model consists of three main components:</p>
<ol type="1">
<li><p><strong>Random component</strong></p>
<p>Specifies the probability distribution of the response variable <span class="math inline">Y</span>. Unlike linear regression, where <span class="math inline">Y</span> is assumed to follow a normal distribution, GLMs allow for different distributions from the exponential family, such as normal distribution, binomial distribution, Poisson distribution, Gamma distribution.</p></li>
<li><p><strong>Systematic component</strong></p>
<p>This refers to the linear predictor, which is the linear combination of the explanatory (independent) variables. It is similar to the equation used in linear regression:</p>
<p><span class="math display">
\eta = \beta_0 + \beta_1 X_1 + \beta_2 X_2 + \dots + \beta_p X_p
</span>where <span class="math inline">\eta</span> is the linear predictor, <span class="math inline">\beta_0, \beta_1, \dots, \beta_p</span> are the regression coefficients, <span class="math inline">X_1, X_2, \dots, X_p</span> are the independent variables.</p></li>
<li><p><strong>Link function</strong></p>
<p>The link function connects the linear predictor <span class="math inline">\eta</span> to the mean of the response variable. This is important because, in many cases, the response variable’s distribution and scale are not appropriate for direct use in a linear equation. The most commonly used link functions are:</p>
<ul>
<li>Identity link (for normal distribution): <span class="math inline">\eta = \mu</span></li>
<li>Logit link (for binomial distribution): <span class="math inline">\eta = \ln \left( \frac{\mu}{1 - \mu} \right)</span></li>
<li>Log link (for Poisson distribution): <span class="math inline">\eta = \ln(\mu)</span></li>
<li>Inverse link (for gamma distribution): <span class="math inline">\eta = \mu^{-1}, \ \text{where} \ \mu</span> is the expected value of <span class="math inline">Y</span>.</li>
</ul>
<p>where <span class="math inline">\mu</span> is the expected value of <span class="math inline">Y</span>.</p></li>
</ol>
<p>The general form of a GLM is:</p>
<p><span class="math display">
\text{g}(\mu_i) = \beta_0 + \beta_1 X_1 + \dots + \beta_p X_p
</span></p>
<p>where <span class="math inline">g(\mu_i)</span> is the link function that relates the expected value <span class="math inline">\mu_i</span> of the response variable <span class="math inline">Y_i</span> to the linear predictor.</p>
<p>Below are some common distributions and their link functions:</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 20%">
<col style="width: 22%">
<col style="width: 34%">
<col style="width: 22%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Distribution</th>
<th style="text-align: left;">Dependent variable</th>
<th style="text-align: left;">Link function</th>
<th style="text-align: left;">Common model</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Normal distribution</td>
<td style="text-align: left;">Continuous data</td>
<td style="text-align: left;">Identity g(<span class="math inline">\mu</span>) = <span class="math inline">\mu</span></td>
<td style="text-align: left;">Linear regression</td>
</tr>
<tr class="even">
<td style="text-align: left;">Binomial distribution</td>
<td style="text-align: left;">Binary data</td>
<td style="text-align: left;">Logit g(<span class="math inline">\mu</span>) = ln(<span class="math inline">\mu / (1 - \mu)</span>)</td>
<td style="text-align: left;">Logistic regression</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Poisson distribution</td>
<td style="text-align: left;">Count data</td>
<td style="text-align: left;">Log g(<span class="math inline">\mu</span>) = ln(<span class="math inline">\mu</span>)</td>
<td style="text-align: left;">Poisson regression</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gamma distribution</td>
<td style="text-align: left;">Positive continuous data</td>
<td style="text-align: left;">Inverse g(<span class="math inline">\mu</span>) = <span class="math inline">1 / \mu</span></td>
<td style="text-align: left;">Gamma regression</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Inverse Gaussian distribution</td>
<td style="text-align: left;">Positive skewed continuous data</td>
<td style="text-align: left;">Inverse squared g(<span class="math inline">\mu</span>) = <span class="math inline">1 / \mu^2</span></td>
<td style="text-align: left;">Inverse Gaussian regression</td>
</tr>
<tr class="even">
<td style="text-align: left;">Multinomial distribution</td>
<td style="text-align: left;">Categorical data</td>
<td style="text-align: left;">Multinomial Logit g(<span class="math inline">\mu</span>) = ln(<span class="math inline">\mu_j / \mu_K</span>)</td>
<td style="text-align: left;">Multinomial logistic regression</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The GLMs are used to model relationships between clinical predictors and outcomes, where the response variable can follow different distributions (e.g., binomial, Poisson). The correct specification of the link function and distribution is critical for drawing valid inferences.</p>
</section>
<section id="log-linear-model" class="level2" data-number="20.3">
<h2 data-number="20.3" class="anchored" data-anchor-id="log-linear-model"><span class="header-section-number">20.3</span> Log-linear model</h2>
<p>Log-linear model is often used to describe the relationship between categorical variables in a contingency table by modeling the natural logarithm of the expected cell frequencies as a linear combination of parameters. This model is often applied to multidimensional categorical data and is particularly useful for exploring interactions between categorical variables.</p>
<p>The general form for a 3-way table with variables <span class="math inline">X_1</span> , <span class="math inline">X_2</span>, and <span class="math inline">X_3</span> is:</p>
<p><span class="math display">
\ln(\mu_{ijk}) = \lambda + \lambda_{i}^{(X_1)} + \lambda_{j}^{(X_2)} + \lambda_{k}^{(X_3)} + \lambda_{ij}^{(X_1 X_2)} + \lambda_{ik}^{(X_1 X_3)} + \lambda_{jk}^{(X_2 X_3)} + \lambda_{ijk}^{(X_1 X_2 X_3)}
</span></p>
<p>where <span class="math inline">\mu_{ijk}</span> is the expected frequency for the combination of category <span class="math inline">i</span>, <span class="math inline">j</span>, and <span class="math inline">k</span>, <span class="math inline">\lambda</span> is an intercept term (overall mean), <span class="math inline">\lambda_{i}^{(X_1)}</span>, <span class="math inline">\lambda_{j}^{(X_2)}</span>, and <span class="math inline">\lambda_{k}^{(X_3)}</span>are the main effects for each variable,<span class="math inline">\lambda_{ij}^{(X_1 X_2)}</span>, <span class="math inline">\lambda_{ik}^{(X_1 X_3)}</span><em>,</em> and <span class="math inline">\lambda_{jk}^{(X_2 X_3)}</span> are the interaction effects between pairs of variables, <span class="math inline">\lambda_{ijk}^{(X_1 X_2 X_3)}</span> is the three-way interaction term among the three variables.</p>
<p>There are several types of log-linear models, depending on the complexity of the relationships they model. These include:</p>
<ol type="1">
<li><p><strong>Independence model</strong></p>
<p>This assumes that all variables are independent of each other, meaning no interaction terms are included. For a three-way table, the model looks like:</p>
<p><span class="math display">
\ln(\mu_{ijk}) = \lambda + \lambda_{i}^{(X_1)} + \lambda_{j}^{(X_2)} + \lambda_{k}^{(X_3)}
</span></p>
<p>This model assumes no association between variables X_1 , X_2 , and X_3 .</p></li>
<li><p><strong>Interaction model</strong></p>
<p>This model includes interactions between two or more variables. For example, a model that includes two-way interactions but excludes higher-order interactions is:</p>
<p><span class="math display">
\ln(\mu_{ijk}) = \lambda + \lambda_{i}^{(X_1)} + \lambda_{j}^{(X_2)} + \lambda_{k}^{(X_3)} + \lambda_{ij}^{(X_1 X_2)} + \lambda_{ik}^{(X_1 X_3)} + \lambda_{jk}^{(X_2 X_3)}
</span>This allows for interaction between pairs of variables but not all three variables together.</p></li>
<li><p><strong>Saturated model</strong></p>
<p>This includes all possible main effects and interactions (up to the highest order), fully describing the relationships between the variables:</p>
<p><span class="math display">
\ln(\mu_{ijk}) = \lambda + \lambda_{i}^{(X_1)} + \lambda_{j}^{(X_2)} + \lambda_{k}^{(X_3)} + \lambda_{ij}^{(X_1 X_2)} + \lambda_{ik}^{(X_1 X_3)} + \lambda_{jk}^{(X_2 X_3)} + \lambda_{ijk}^{(X_1 X_2 X_3)}
</span></p>
<p>This model perfectly fits the observed data but might overfit if the data set is small.</p></li>
</ol>
<p>These types of log-linear models allow flexibility in modeling categorical data, from simple independence models to complex interaction models depending on the structure and hypotheses in the data.</p>
<p>Log-linear models have several key applications, particularly in the analysis of categorical data in contingency tables.</p>
<ol type="1">
<li><strong>Contingency table analysis</strong>: Log-linear models are commonly used to analyze contingency tables, where they provide a way to model and test for associations between multiple categorical variables.</li>
<li><strong>Interaction analysis</strong>: Log-linear models are ideal for identifying interaction effects between multiple categorical variables. For example, they can help understand if a response variable depends on combinations of two or more predictors.</li>
<li><strong>Association testing</strong>: Log-linear models allow testing hypothesis about the independence or association between categorical variables in multi-way tables.</li>
</ol>
<p>Researchers often fit both saturated (fully interactive) and reduced models to identify which interactions and main effects are statistically significant. The interpretation is based on the estimated parameters, which represent the log of the odds ratios. Understanding these parameters is crucial in drawing meaningful conclusions about the relationships between variables.</p>
<p>Log-linear models are powerful tools for exploring multi-way relationships between categorical variables, making them widely applicable across various fields of research.</p>
<section id="goodness-of-fit-test" class="level3" data-number="20.3.1">
<h3 data-number="20.3.1" class="anchored" data-anchor-id="goodness-of-fit-test"><span class="header-section-number">20.3.1</span> Goodness-of-fit test</h3>
<p>The goodness-of-fit test in log-linear model is used to assess how well the model fits the observed data. Two main tests are typically used for this purpose:</p>
<ol type="1">
<li><p><strong>Likelihood ratio test</strong></p>
<p>The likelihood ratio test (also called the deviance test) compares the deviance of a fitted log-linear model to the deviance of a saturated model (a model with as many parameters as data points). The deviance <span class="math inline">D</span> is calculated as:</p>
<p><span class="math display">
D = 2 \sum O \cdot \ln \left( \frac{O}{E} \right)
</span></p>
<p>where <span class="math inline">O</span> are the observed cell frequencies, and <span class="math inline">E</span> are the expected cell frequencies under the model.</p>
<p>The deviance follows a chi-square distribution with degrees of freedom equal to the difference in the number of parameters between the two models. If the p-value is small (e.g., less than 0.05), the null hypothesis is rejected, indicating that the model does not fit the data well.</p></li>
<li><p><strong>Pearson chi-square test</strong></p>
<p>The Pearson chi-square test is an alternative goodness-of-fit test for log-linear models, which measures the discrepancy between observed and expected frequencies. The test statistic is:</p>
<p><span class="math display">
\chi^2 = \sum \frac{(O - E)^2}{E}
</span></p>
<p>Like the deviance test, the Pearson chi-square statistic follows a chi-square distribution with degrees of freedom based on the number of cells and the number of parameters. If the p-value from the chi-square test is significant, the model does not provide a good fit.</p></li>
</ol>
</section>
<section id="parameter-estimation" class="level3" data-number="20.3.2">
<h3 data-number="20.3.2" class="anchored" data-anchor-id="parameter-estimation"><span class="header-section-number">20.3.2</span> Parameter estimation</h3>
<p>The parameters are usually estimated using maximum likelihood estimation (MLE). This method finds the parameter values that maximize the likelihood function <span class="math inline">L(\beta)</span>, which represents the probability of observing the given data under the model. The log-likelihood function is often maximized instead of the likelihood because it is easier to work with sums of logs rather than products. In practice, MLE is often solved using iterative algorithms like iteratively reweighted least squares or Newton-Raphson.</p>
<ul>
<li><p><strong>Interpretation of parameters</strong></p>
<p>The coefficients <span class="math inline">\lambda</span> reflect the log odds ratios for the associations between the variables. If the interaction terms are significant, it indicates that there are interactions between the variables.</p>
<p>A positive parameter estimate for an interaction term (e.g., <span class="math inline">\lambda^{XY}</span>) suggests that the odds of the joint occurrence of <span class="math inline">X</span> and <span class="math inline">Y</span> increase relative to the marginal effects. A negative estimate indicates a negative association between the variables.</p></li>
<li><p><strong>Testing parameter significance</strong></p>
<p>The significance of individual parameters can be tested using Wald tests or likelihood ratio tests. The Wald test assesses whether each estimated coefficient is significantly different from zero:</p>
<p><span class="math display">
z = \frac{\hat{\beta}}{\text{SE}(\hat{\beta})}
</span></p>
<p>The test statistic follows a standard normal distribution, and a significant test indicates that the parameter contributes significantly to the model.</p></li>
</ul>
<div class="example">
<strong>Example 1</strong> 
<p>A study employed a case-control design to investigate the impact of maternal education level on growth and developmental delays in children under the age of 5. A total of 173 cases (children with developmental delays) and 173 controls (children with normal development) were surveyed, including their mothers’ education levels.</p>
</div>
<div>
<p><a href="datasets/ex20-01.csv" download="ex20-01.csv" class="btn btn-success"> <i class="bi bi-database-fill-down"></i> Download data </a></p>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>tb <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"datasets/ex20-01.csv"</span>, <span class="at">show_col_types =</span> F) <span class="sc">|&gt;</span> </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">edu =</span> <span class="fu">factor</span>(edu),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">grow =</span> <span class="fu">factor</span>(grow),</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">edu =</span> <span class="fu">relevel</span>(edu, <span class="at">ref =</span> <span class="dv">3</span>),</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">grow =</span> <span class="fu">relevel</span>(grow, <span class="at">ref =</span> <span class="dv">2</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>You can fit log-linear models using the <code>loglm()</code> function from the <code>MASS</code> package or the <code>glm()</code> function with a <code>Poisson</code> family specification.</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-3"><pre class="sourceCode r code-annotation-code code-with-copy code-annotated"><code class="sourceCode r"><span id="annotated-cell-3-1"><a href="#annotated-cell-3-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">loglm</span>(f <span class="sc">~</span> edu <span class="sc">+</span> grow, <span class="at">data =</span> tb)</span>
<span id="annotated-cell-3-2"><a href="#annotated-cell-3-2" aria-hidden="true" tabindex="-1"></a>fit1 <span class="ot">&lt;-</span> <span class="fu">update</span>(fit, .<span class="sc">~</span>.<span class="sc">^</span><span class="dv">2</span>)</span>
<button class="code-annotation-anchor" data-target-cell="annotated-cell-3" data-target-annotation="1">1</button><span id="annotated-cell-3-3" class="code-annotation-target"><a href="#annotated-cell-3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#fit2 &lt;- step(fit1, direction = "backward")</span></span>
<span id="annotated-cell-3-4"><a href="#annotated-cell-3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(fit, fit1)</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-hidden code-annotation-container-grid">
<dt data-target-cell="annotated-cell-3" data-target-annotation="1">1</dt>
<dd>
<span data-code-cell="annotated-cell-3" data-code-lines="3" data-code-annotation="1">The <code>step()</code> may not work properly with <code>loglm()</code> object when there is only when independent variable left in the model. However, <code>glm()</code> object does.</span>
</dd>
</dl>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; LR tests for hierarchical log-linear models
#&gt; 
#&gt; Model 1:
#&gt;  f ~ edu + grow 
#&gt; Model 2:
#&gt;  f ~ edu + grow + edu:grow 
#&gt; 
#&gt;            Deviance df Delta(Dev) Delta(df) P(&gt; Delta(Dev)
#&gt; Model 1   0.4585613  2                                    
#&gt; Model 2   0.0000000  0  0.4585613         2        0.79511
#&gt; Saturated 0.0000000  0  0.0000000         0        1.00000</code></pre>
</div>
</div>
<p>The results show that the p-value for the likelihood ratio test is 0.79511, which is much larger than the typical significance level (0.05). This shows there is no significant improvement in fit by adding the interaction term (<code>edu:grow</code>) to the model. Therefore, the simpler <em>Model 1</em> without the interaction is sufficient, and there is no strong evidence that the interaction between maternal education and child growth is important for explaining the data.</p>
<p>Thus, the simpler model (main effects only) is preferred because the interaction term does not significantly improve the model fit.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>glm_fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(f <span class="sc">~</span> edu <span class="sc">+</span> grow, <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>), <span class="at">data =</span> tb)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>glm_fit1 <span class="ot">&lt;-</span> <span class="fu">update</span>(glm_fit, .<span class="sc">~</span>.<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>glm_fit2 <span class="ot">&lt;-</span> <span class="fu">step</span>(glm_fit1, <span class="at">direction =</span> <span class="st">"backward"</span>, <span class="at">trace =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; Start:  AIC=47.08
#&gt; f ~ edu + grow + edu:grow
#&gt; 
#&gt;            Df Deviance    AIC
#&gt; - edu:grow  2  0.45856 43.536
#&gt; &lt;none&gt;         0.00000 47.078
#&gt; 
#&gt; Step:  AIC=43.54
#&gt; f ~ edu + grow
#&gt; 
#&gt;        Df Deviance    AIC
#&gt; - grow  1    0.459 41.536
#&gt; &lt;none&gt;       0.459 43.536
#&gt; - edu   2   34.927 74.005
#&gt; 
#&gt; Step:  AIC=41.54
#&gt; f ~ edu
#&gt; 
#&gt;        Df Deviance    AIC
#&gt; &lt;none&gt;       0.459 41.536
#&gt; - edu   2   34.927 72.005</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(glm_fit2, glm_fit, glm_fit1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; Analysis of Deviance Table
#&gt; 
#&gt; Model 1: f ~ edu
#&gt; Model 2: f ~ edu + grow
#&gt; Model 3: f ~ edu + grow + edu:grow
#&gt;   Resid. Df Resid. Dev Df Deviance Pr(&gt;Chi)
#&gt; 1         3    0.45856                     
#&gt; 2         2    0.45856  1  0.00000         
#&gt; 3         0    0.00000  2  0.45856   0.7951</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Results
</div>
</div>
<div class="callout-body-container callout-body">
<p>There is no significant improvement in model fit when adding grow (Model 2) or the interaction term edu:grow (Model 3). Model 1 (with only edu) may be sufficient, as adding more terms does not significantly enhance the fit.</p>
</div>
</div>
<p>In this context, the p-value is 0.7951, same to that from the likelihood ratio test made above.</p>
<p>The parameters of the final fitting model can be retrieved using the code below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>glm_fit2 <span class="sc">|&gt;</span> <span class="fu">coef</span>()  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; (Intercept)        edu1        edu2 
#&gt;   3.8918203   0.5330263  -0.1905183</code></pre>
</div>
</div>
<p>The regression equation is as follows:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>equatiomatic<span class="sc">::</span><span class="fu">extract_eq</span>(glm_fit2, <span class="at">use_coefs =</span> T, <span class="at">coef_digits =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><span class="math display">
\log ({ \widehat{E( \operatorname{f} )} })  = 3.89182 + 0.53303(\operatorname{edu}_{\operatorname{1}}) - 0.19052(\operatorname{edu}_{\operatorname{2}})
</span></p>
</div>
</div>
<div class="example">
<strong>Example 2</strong> 
<p>A case-control study was conducted to investigate the role of contraceptives and the allele of clotting factor V-Leiden in the development of venous thrombosis. A total of 324 people were investigated, including 155 cases and 169 controls. Analyze the interaction between contraceptive use and the gene.</p>
</div>
<div>
<p><a href="datasets/ex20-02.csv" download="ex20-02.csv" class="btn btn-success"> <i class="bi bi-database-fill-down"></i> Download data </a></p>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>tb <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"datasets/ex20-02.csv"</span>, <span class="at">show_col_types =</span> F) <span class="sc">|&gt;</span> </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">grp =</span> <span class="fu">factor</span>(grp),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">expo =</span> <span class="fu">factor</span>(expo),</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">gtype =</span> <span class="fu">factor</span>(gtype),</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">grp =</span> <span class="fu">relevel</span>(grp, <span class="at">ref =</span> <span class="dv">2</span>),</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">expo =</span> <span class="fu">relevel</span>(expo, <span class="at">ref =</span> <span class="dv">2</span>),</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">gtype =</span> <span class="fu">relevel</span>(gtype, <span class="at">ref =</span> <span class="dv">2</span>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">loglm</span>(f <span class="sc">~</span> grp <span class="sc">+</span> expo <span class="sc">+</span> gtype, <span class="at">data =</span> tb)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>fit1 <span class="ot">&lt;-</span> <span class="fu">update</span>(fit, .<span class="sc">~</span>.<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">update</span>(fit, .<span class="sc">~</span>.<span class="sc">^</span><span class="dv">3</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>fit3 <span class="ot">&lt;-</span> <span class="fu">step</span>(fit2, <span class="at">direction =</span> <span class="st">"backward"</span>, <span class="at">trace =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; Start:  AIC=16
#&gt; f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype + 
#&gt;     grp:expo:gtype
#&gt; 
#&gt;                  Df    AIC
#&gt; - grp:expo:gtype  1 14.096
#&gt; &lt;none&gt;              16.000
#&gt; 
#&gt; Step:  AIC=14.1
#&gt; f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype
#&gt; 
#&gt;              Df    AIC
#&gt; - expo:gtype  1 12.097
#&gt; &lt;none&gt;          14.096
#&gt; - grp:gtype   1 37.909
#&gt; - grp:expo    1 42.920
#&gt; 
#&gt; Step:  AIC=12.1
#&gt; f ~ grp + expo + gtype + grp:expo + grp:gtype
#&gt; 
#&gt;             Df    AIC
#&gt; &lt;none&gt;         12.097
#&gt; - grp:gtype  1 38.751
#&gt; - grp:expo   1 43.762</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(fit, fit1, fit2, fit3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; LR tests for hierarchical log-linear models
#&gt; 
#&gt; Model 1:
#&gt;  f ~ grp + expo + gtype 
#&gt; Model 2:
#&gt;  f ~ grp + expo + gtype + grp:expo + grp:gtype 
#&gt; Model 3:
#&gt;  f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype 
#&gt; Model 4:
#&gt;  f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype + grp:expo:gtype 
#&gt; 
#&gt;              Deviance df   Delta(Dev) Delta(df) P(&gt; Delta(Dev)
#&gt; Model 1   62.41639390  4                                      
#&gt; Model 2    0.09702652  2 6.231937e+01         2        0.00000
#&gt; Model 3    0.09613097  1 8.955578e-04         1        0.97613
#&gt; Model 4    0.00000000  0 9.613097e-02         1        0.75652
#&gt; Saturated  0.00000000  0 0.000000e+00         0        1.00000</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>glm_fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(f <span class="sc">~</span> grp <span class="sc">+</span> expo <span class="sc">+</span> gtype, <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>), <span class="at">data =</span> tb)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>glm_fit1 <span class="ot">&lt;-</span> <span class="fu">update</span>(glm_fit, .<span class="sc">~</span>.<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>glm_fit2 <span class="ot">&lt;-</span> <span class="fu">update</span>(glm_fit, .<span class="sc">~</span>.<span class="sc">^</span><span class="dv">3</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>glm_fit3 <span class="ot">&lt;-</span> <span class="fu">step</span>(glm_fit2, <span class="at">direction =</span> <span class="st">"backward"</span>, <span class="at">trace =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; Start:  AIC=55.23
#&gt; f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype + 
#&gt;     grp:expo:gtype
#&gt; 
#&gt;                  Df Deviance    AIC
#&gt; - grp:expo:gtype  1 0.096129 53.321
#&gt; &lt;none&gt;              0.000000 55.225
#&gt; 
#&gt; Step:  AIC=53.32
#&gt; f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype
#&gt; 
#&gt;              Df Deviance    AIC
#&gt; - expo:gtype  1   0.0970 51.322
#&gt; &lt;none&gt;            0.0961 53.321
#&gt; - grp:gtype   1  25.9093 77.134
#&gt; - grp:expo    1  30.9197 82.145
#&gt; 
#&gt; Step:  AIC=51.32
#&gt; f ~ grp + expo + gtype + grp:expo + grp:gtype
#&gt; 
#&gt;             Df Deviance    AIC
#&gt; &lt;none&gt;            0.097 51.322
#&gt; - grp:gtype  1   28.751 77.977
#&gt; - grp:expo   1   33.762 82.987</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(glm_fit, glm_fit3, glm_fit1, glm_fit2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; Analysis of Deviance Table
#&gt; 
#&gt; Model 1: f ~ grp + expo + gtype
#&gt; Model 2: f ~ grp + expo + gtype + grp:expo + grp:gtype
#&gt; Model 3: f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype
#&gt; Model 4: f ~ grp + expo + gtype + grp:expo + grp:gtype + expo:gtype + 
#&gt;     grp:expo:gtype
#&gt;   Resid. Df Resid. Dev Df Deviance  Pr(&gt;Chi)    
#&gt; 1         4     62.416                          
#&gt; 2         2      0.097  2   62.319 2.934e-14 ***
#&gt; 3         1      0.096  1    0.001    0.9761    
#&gt; 4         0      0.000  1    0.096    0.7565    
#&gt; ---
#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Results
</div>
</div>
<div class="callout-body-container callout-body">
<p>Model 2 shows a significant improvement over Model 1, suggesting that the interaction terms grp:expo and grp:gtype significantly improve the model’s fit. Model 3 and Model 4 show no significant improvement over Model 2, with p-values above 0.75, meaning that adding the interaction term expo:gtype or the three-way interaction does not significantly improve the fit.</p>
<p>Thus, Model 2 is the most appropriate model, as it includes the important two-way interaction terms and significantly improves the fit without overcomplicating the model.</p>
</div>
</div>
<p>The regression coefficients of the final fitting model is:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>opt_model <span class="ot">&lt;-</span> <span class="fu">glm</span>(</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  f <span class="sc">~</span> grp <span class="sc">*</span> expo <span class="sc">*</span> gtype, </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>), </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> tb) <span class="sc">|&gt;</span> </span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step</span>(<span class="at">direction =</span> <span class="st">"backward"</span>, <span class="at">trace =</span> <span class="dv">0</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(opt_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; (Intercept)        grp1       expo1      gtype1  grp1:expo1 grp1:gtype1 
#&gt;   4.6082424  -1.0355344  -0.4700036  -3.3019907   1.3327101   2.0698471</code></pre>
</div>
</div>
<p>The regression equation is as follows:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>equatiomatic<span class="sc">::</span><span class="fu">extract_eq</span>(opt_model, <span class="at">use_coefs =</span> T, <span class="at">coef_digits =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><span class="math display">
\log ({ \widehat{E( \operatorname{f} )} })  = 4.60824 - 1.03553(\operatorname{grp}_{\operatorname{1}}) - 0.47(\operatorname{\expo}_{\operatorname{1}}) - 3.30199(\operatorname{gtype}_{\operatorname{1}}) + 1.33271(\operatorname{grp}_{\operatorname{1}} \times \operatorname{\expo}_{\operatorname{1}}) + 2.06985(\operatorname{grp}_{\operatorname{1}} \times \operatorname{gtype}_{\operatorname{1}})
</span></p>
</div>
</div>
<p>Residual analysis helps in assessing whether your model fits the data well and whether the assumptions of the model are met. Here’s how you can conduct residual analysis for your generalized linear model:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">dev_resid =</span> <span class="fu">residuals</span>(opt_model, <span class="at">type =</span> <span class="st">"deviance"</span>),</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">fit_value =</span> <span class="fu">fitted.values</span>(opt_model))</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot residuals vs fitted values</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(fit_value, dev_resid)) <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"red"</span>)<span class="sc">+</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Fitted values"</span>, </span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Deviance residuals"</span>,</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Residuals plot"</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="generalized-linear-model_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Q-Q plot</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">sample =</span> dev_resid)) <span class="sc">+</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_qq</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_qq_line</span>() <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Theoretical quantiles"</span>, <span class="at">y =</span> <span class="st">"Sample quantiles"</span>, <span class="at">title =</span> <span class="st">"Q-Q plot"</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Leverage plot</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="fu">tibble</span>(</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">hat_value =</span> <span class="fu">hatvalues</span>(opt_model),</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">index =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hat_value))) <span class="sc">|&gt;</span> </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(index, hat_value)) <span class="sc">+</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Index"</span>, <span class="at">y =</span> <span class="st">"Hat_values"</span>, <span class="at">title =</span> <span class="st">"Leverage plot"</span>)</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Cook's distance</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="fu">tibble</span>(</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">cooks_dis =</span> <span class="fu">cooks.distance</span>(opt_model),</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">index =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(cooks_dis))) <span class="sc">|&gt;</span> </span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(index, cooks_dis)) <span class="sc">+</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">4</span><span class="sc">/</span>(<span class="fu">nrow</span>(df)<span class="sc">-</span><span class="fu">length</span>(<span class="fu">coef</span>(opt_model))), <span class="at">col =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Index"</span>, <span class="at">y =</span> <span class="st">"Cook's distance"</span>, <span class="at">title =</span> <span class="st">"Cook's distance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="generalized-linear-model_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="generalized-linear-model_files/figure-html/unnamed-chunk-15-2.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="generalized-linear-model_files/figure-html/unnamed-chunk-15-3.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Interpretation
</div>
</div>
<div class="callout-body-container callout-body">
<p>If the residuals are randomly scattered around zero in the Residuals plot, and the Q-Q plot follows a straight line, your model fits well.</p>
<p>Observations with high leverage or high Cook’s distance might indicate influential points. You may want to examine these data points closely.</p>
<p>This analysis helps assess the final model’s fit and identify any potential issues such as outliers or non-linearity.</p>
</div>
</div>
</section>
</section>
<section id="poisson-regression" class="level2" data-number="20.4">
<h2 data-number="20.4" class="anchored" data-anchor-id="poisson-regression"><span class="header-section-number">20.4</span> Poisson regression</h2>
<p>Poisson regression is a type of generalized linear model used for count data. It is particularly useful when the dependent variable represents counts of events that occur independently over a fixed period of time or space. The model assumes that the count data follows a Poisson distribution.</p>
<p>The natural logarithm is used as the link function, connecting the linear predictor to the expected value of the response variable. This means that the model predicts the logarithm of the expected count. It assumes that the events are independent, and the mean and variance of the counts are equal.</p>
<p>The general form of a Poisson regression model can be expressed as:</p>
<p><span class="math display">
\ln(\lambda_i) = \beta_0 + \beta_1 X_{i1} + \beta_2 X_{i2} + \ldots + \beta_k X_{ik}
</span></p>
<p>where <span class="math inline">\lambda</span> is the expected count for the <span class="math inline">i</span>-th observation, <span class="math inline">X_{i1}, X_{i2}, \ldots, X_{ik}</span> are the predictor variables, <span class="math inline">\beta_0, \beta_1, \ldots, \beta_k</span> are the coefficients to be estimated.</p>
<p>A Poisson regression model with an offset is used when the rate of occurrence of events is being modeled, rather than the raw counts. The offset allows the model to account for different exposure times or different sizes of population at risk. In this case, the model becomes:</p>
<p><span class="math display">
\ln(\lambda_i) = \beta_0 + \beta_1 X_{i1} + \beta_2 X_{i2} + \ldots + \beta_p X_{ik} + \log(t_i)
</span></p>
<p>where <span class="math inline">t_i</span> is the offset (often the exposure time or population size for each observation). The offset is added to the linear predictor as <span class="math inline">\log(t_i)</span>, and it is treated as a known value rather than a variable for which a coefficient is estimated.</p>
<p>This adjustment is essential when the time or population at risk differs across observations.</p>
<section id="goodness-of-fit-test-1" class="level3" data-number="20.4.1">
<h3 data-number="20.4.1" class="anchored" data-anchor-id="goodness-of-fit-test-1"><span class="header-section-number">20.4.1</span> Goodness-of-fit test</h3>
<p>Goodness-of-fit tests are used to assess how well the model fits the data. Here are some common methods to conduct goodness-of-fit tests for Poisson regression.</p>
<p><strong>Deviance and Pearson tests</strong></p>
<p>Deviance is the most commonly used goodness-of-fit statistic in Poisson regression. It compares the model’s deviance to its degrees of freedom to assess the fit.</p>
<p>Pearson test involves calculating the sum of squared Pearson residuals and comparing it to its degrees of freedom.</p>
<p><strong>Likelihood ratio test</strong></p>
<p>This test compares the likelihood functions of the full model and a reduced model to determine whether adding significant variables improves the model fit.</p>
</section>
<section id="parameters-estimation" class="level3" data-number="20.4.2">
<h3 data-number="20.4.2" class="anchored" data-anchor-id="parameters-estimation"><span class="header-section-number">20.4.2</span> Parameters estimation</h3>
<p>The parameters are estimated by maximizing the likelihood function, which for the Poisson distribution is given by:</p>
<p><span class="math display">
L(\beta) = \prod_{i=1}^n \frac{e^{-\lambda_i} \lambda_i^{Y_i}}{Y_i!}
</span></p>
<p>To simplify the computation, the log-likelihood function is typically used. It is written as:</p>
<p><span class="math display">
\log L(\beta) = \sum_{i=1}^{n} \left( Y_i \log(\lambda_i) - \lambda_i - \log(Y_i!) \right)
</span></p>
<p>Substituting <span class="math inline">\lambda_i = e^{X_i \beta}</span> into the equation, we get:</p>
<p><span class="math display">
\log L(\beta) = \sum_{i=1}^{n} \left( Y_i (X_i \beta) - e^{X_i \beta} - \log(Y_i!) \right)
</span></p>
<p>where <span class="math inline">\lambda_i = e^{X_i \beta}</span>, representing the expected count for observation <span class="math inline">i</span>, determined by the independent variables <span class="math inline">X_i</span> and regression coefficients <span class="math inline">\beta</span>, <span class="math inline">Y_i</span> is the observed count for the <span class="math inline">i</span>-th observation.</p>
<p>By maximizing this log-likelihood function, the regression coefficients <span class="math inline">\beta</span> can be estimated for the Poisson regression model.</p>
<p>To assess whether a predictor is statistically significant, calculate the <span class="math inline">Z</span> statistic:</p>
<p><span class="math display">
Z = \frac{\hat{\beta}}{\text{SE}(\hat{\beta})}
</span></p>
<p>where <span class="math inline">\hat{\beta}</span> is the estimated coefficient and <span class="math inline">\text{SE}(\hat{\beta})</span> is the standard error. The p-value is obtained from the standard normal distribution based on the calculated z-value. If the p-value is less than the significance level (commonly 0.05), reject the null hypothesis, indicating that the predictor is statistically significant.</p>
<div class="example">
<strong>Example 3</strong> 
<p>A researcher conducted a retrospective cohort study to investigate the relationship between arsenic exposure in a smelting plant and deaths due to respiratory diseases among the plant’s employees from 1978 to 2009. The data can be download below. Please analyze this data.</p>
</div>
<div>
<p><a href="datasets/ex20-03.csv" download="ex20-03.csv" class="btn btn-success"> <i class="bi bi-database-fill-down"></i> Download data </a></p>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>tb <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"datasets/ex20-03.csv"</span>, <span class="at">show_col_types =</span> F) <span class="sc">|&gt;</span> </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">AsExpo =</span> <span class="fu">factor</span>(AsExpo, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"As.No"</span>, <span class="st">"As.Yes"</span>)),</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">ageGroup =</span> <span class="fu">factor</span>(ageGroup, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"40~49"</span>, <span class="st">"50~59"</span>, <span class="st">"60~69"</span>, <span class="st">"&gt;=70"</span>)),</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">nDeath =</span> <span class="fu">as.integer</span>(nDeath),</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">nTotal =</span> <span class="fu">as.integer</span>(<span class="fu">round</span>(nTotal))</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>tb <span class="sc">|&gt;</span> </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from  =</span> AsExpo,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> <span class="fu">c</span>(nDeath, nTotal)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span> </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="at">align =</span> <span class="st">"c"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 14%">
<col style="width: 22%">
<col style="width: 20%">
<col style="width: 22%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">ageGroup</th>
<th style="text-align: center;">nDeath_As.Yes</th>
<th style="text-align: center;">nDeath_As.No</th>
<th style="text-align: center;">nTotal_As.Yes</th>
<th style="text-align: center;">nTotal_As.No</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">40~49</td>
<td style="text-align: center;">7</td>
<td style="text-align: center;">14</td>
<td style="text-align: center;">11026</td>
<td style="text-align: center;">38337</td>
</tr>
<tr class="even">
<td style="text-align: center;">50~59</td>
<td style="text-align: center;">42</td>
<td style="text-align: center;">38</td>
<td style="text-align: center;">10792</td>
<td style="text-align: center;">31019</td>
</tr>
<tr class="odd">
<td style="text-align: center;">60~69</td>
<td style="text-align: center;">59</td>
<td style="text-align: center;">58</td>
<td style="text-align: center;">6898</td>
<td style="text-align: center;">17496</td>
</tr>
<tr class="even">
<td style="text-align: center;">&gt;=70</td>
<td style="text-align: center;">17</td>
<td style="text-align: center;">41</td>
<td style="text-align: center;">2581</td>
<td style="text-align: center;">6842</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>You can use the <code>glm()</code> function to fit a Poisson regression model.</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-19"><pre class="sourceCode r code-annotation-code code-with-copy"><code class="sourceCode r"><span id="annotated-cell-19-1"><a href="#annotated-cell-19-1" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">glm</span>(</span>
<span id="annotated-cell-19-2"><a href="#annotated-cell-19-2" aria-hidden="true" tabindex="-1"></a>  nDeath <span class="sc">~</span> AsExpo <span class="sc">+</span> ageGroup, </span>
<span id="annotated-cell-19-3"><a href="#annotated-cell-19-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>),</span>
<span id="annotated-cell-19-4"><a href="#annotated-cell-19-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> tb, </span>
<span id="annotated-cell-19-5"><a href="#annotated-cell-19-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">offset =</span> <span class="fu">log</span>(nTotal)</span>
<span id="annotated-cell-19-6"><a href="#annotated-cell-19-6" aria-hidden="true" tabindex="-1"></a>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Offset
</div>
</div>
<div class="callout-body-container callout-body">
<p>The inclusion of the <code>log(nTotal)</code> as an offset allows us to model the mortality rate per person-year rather than just the raw number of deaths.</p>
<p>Without the offset, the model would compare the total number of deaths, which could lead to misleading conclusions if one age group simply had more number of deaths or more years than another. By using person-years as an offset, we adjust for this variability and focus on the mortality rate, making the comparison more meaningful.</p>
<p>This method is commonly used in epidemiology to model incidence rates for diseases, infections, or other health outcomes while adjusting for varying follow-up times or population sizes.</p>
</div>
</div>
<p>The <code>coef()</code> function can print the regression coefficients:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt;   (Intercept)  AsExpoAs.Yes ageGroup50~59 ageGroup60~69  ageGroup&gt;=70 
#&gt;    -8.0086462     0.8108583     1.4701581     2.3660912     2.6237690</code></pre>
</div>
</div>
<p>The regression equation is:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>equatiomatic<span class="sc">::</span><span class="fu">extract_eq</span>(model, <span class="at">use_coefs =</span> T, <span class="at">coef_digits =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><span class="math display">
\log ({ \widehat{E( \operatorname{nDeath} )} })  = -8.00865 + 0.81086(\operatorname{AsExpo}_{\operatorname{As.Yes}}) + 1.47016(\operatorname{ageGroup}_{\operatorname{50\char`\~59}}) + 2.36609(\operatorname{ageGroup}_{\operatorname{60\char`\~69}}) + 2.62377(\operatorname{ageGroup}_{\operatorname{&gt;=70}}) + \operatorname{(offset)}
</span></p>
</div>
</div>
<p>Use the <code>summary()</code> function to evaluate the parameter estimates, significance of predictors, and goodness-of-fit.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; 
#&gt; Call:
#&gt; glm(formula = nDeath ~ AsExpo + ageGroup, family = poisson(link = "log"), 
#&gt;     data = tb, offset = log(nTotal))
#&gt; 
#&gt; Coefficients:
#&gt;               Estimate Std. Error z value Pr(&gt;|z|)    
#&gt; (Intercept)    -8.0086     0.2233 -35.859  &lt; 2e-16 ***
#&gt; AsExpoAs.Yes    0.8109     0.1210   6.699 2.09e-11 ***
#&gt; ageGroup50~59   1.4702     0.2453   5.994 2.04e-09 ***
#&gt; ageGroup60~69   2.3661     0.2372   9.976  &lt; 2e-16 ***
#&gt; ageGroup&gt;=70    2.6238     0.2548  10.297  &lt; 2e-16 ***
#&gt; ---
#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#&gt; 
#&gt; (Dispersion parameter for poisson family taken to be 1)
#&gt; 
#&gt;     Null deviance: 260.9311  on 7  degrees of freedom
#&gt; Residual deviance:   9.9323  on 3  degrees of freedom
#&gt; AIC: 61.344
#&gt; 
#&gt; Number of Fisher Scoring iterations: 4</code></pre>
</div>
</div>
<p>This Poisson regression model analyzes the relationship between arsenic exposure (AsExpo), age groups (ageGroup), and the number of deaths from respiratory diseases (nDeath). The model includes an offset (nTotal), which represents the total number of individuals at risk or the total exposure time, to adjust for varying group sizes or follow-up durations.</p>
<p>Here’s a breakdown of the results:</p>
<ol type="1">
<li><p><strong>Coefficients</strong></p>
<ul>
<li>(Intercept): -8.0086: The intercept represents the baseline log mortality rate for the reference group, which in this case is those not exposed to arsenic (AsExpo = “No”) and in the &lt;50 age group. The estimate of -8.0086 suggests a very low baseline mortality rate when exponentiated.</li>
<li>AsExpoYes: 0.8109: The coefficient for arsenic exposure (Yes) indicates that the log mortality rate is increased by 0.8109 units for those exposed to arsenic compared to those not exposed. Exponentiating this value (<span class="math inline">\exp(0.8109) = 2.25</span>) suggests that arsenic exposure increases the mortality rate by about 125% compared to those unexposed.</li>
<li>ageGroup50~59: 1.4702: For individuals aged 50-59, the log mortality rate increases by 1.4702 units compared to those under 50. Exponentiating this value (<span class="math inline">\exp(1.4702) = 4.35</span>) shows that the mortality rate is approximately 4.35 times higher for this age group compared to those under 50.</li>
<li>ageGroup60~69: 2.3661: For individuals aged 60-69, the log mortality rate increases by 2.3661 units. Exponentiating this value (<span class="math inline">\exp(2.3661) = 10.65</span>) indicates that the mortality rate is about 10.65 times higher for this age group compared to the &lt;50 group.</li>
<li>ageGroup&gt;=70: 2.6238: For individuals aged 70 or older, the log mortality rate increases by 2.6238 units. Exponentiating this value (<span class="math inline">\exp(2.6238) = 13.78</span>) indicates that the mortality rate is 13.78 times higher for this age group compared to the reference (&lt;50) group.</li>
</ul></li>
<li><p><strong>Statistical significance</strong></p>
<ul>
<li>The p-values for all variables are very small (&lt;0.001), indicating that arsenic exposure and all age groups are significantly associated with increased mortality rates.</li>
</ul></li>
<li><p><strong>Model fit</strong></p>
<ul>
<li>Residual Deviance: 9.9323 on 3 degrees of freedom. This indicates how well the model fits the data. The small residual deviance relative to the degrees of freedom suggests a good fit, meaning the model explains most of the variability in the data.</li>
<li>Null Deviance: 260.9311 on 7 degrees of freedom. The null deviance represents the deviance of a model with no predictors, just an intercept. A large difference between null and residual deviance indicates that the predictors significantly improve the model fit.</li>
<li>The AIC (61.344) is a measure of model quality, with lower values indicating a better fit. It’s useful when comparing models to select the best one.</li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Key findings
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Arsenic exposure significantly increases the mortality rate (about 125% increase).</li>
<li>Age is also a strong predictor, with older age groups showing exponentially higher mortality rates.</li>
<li>The model fits the data well, as evidenced by the low residual deviance and significant predictors.</li>
</ul>
</div>
</div></li>
</ol>
<p>If the residual deviance is significantly larger than the degrees of freedom, it may indicate overdispersion, suggesting that the Poisson model may not be appropriate. In such cases, consider using a negative binomial regression or adjusting the model accordingly.</p>
</section>
</section>
<section id="negative-binomial-regression" class="level2" data-number="20.5">
<h2 data-number="20.5" class="anchored" data-anchor-id="negative-binomial-regression"><span class="header-section-number">20.5</span> Negative binomial regression</h2>
<p>In Poisson regression, the assumption is that the mean (<span class="math inline">\lambda</span>) and variance of the dependent variable are equal. When this assumption is violated (i.e., the data is overdispersed), the model may underestimate the standard errors, leading to misleading p-values and confidence intervals.</p>
<p>Negative binomial regression is an extension of Poisson regression that is used when the data exhibits overdispersion—meaning the variance is larger than the mean. The negative binomial regression introduces an extra parameter, called the dispersion parameter or overdispersion parameter, to account for the excess variability.</p>
<ul>
<li>Poisson regression: <span class="math inline">\text{var}(Y) = \mu</span></li>
<li>Negative binomial regression: <span class="math inline">\text{var}(Y) = \mu + \frac{\mu^2}{\theta}</span>,</li>
</ul>
<p>where <span class="math inline">\theta</span> is the dispersion parameter (overdispersion parameter). When <span class="math inline">\theta \to \infty</span>, the negative binomial model reduces to a Poisson model.</p>
<p>The mean of the response variable <span class="math inline">Y_i</span> is related to the predictors <span class="math inline">X_{i1}, X_{i2}, \dots, X_{ik}</span> through the following log-linear relationship:</p>
<p><span class="math display">
\log(\mu_i) = \beta_0 + \beta_1 X_{i1} + \beta_2 X_{i2} + \dots + \beta_k X_{ik}
</span></p>
<p>where <span class="math inline">\mu_i</span> is the expected count (the mean of <span class="math inline">Y_i</span>) for the <span class="math inline">i</span>-th observation, <span class="math inline">X_{i1}, X_{i2}, \dots, X_{ik}</span> are the predictor variables, <span class="math inline">\beta_0, \beta_1, \beta_2, \dots, \beta_k</span> are the coefficients estimated from the data.</p>
<div class="example">
<strong>Example 4</strong> 
<p>A researcher conducted a study to investigate the relationship between types of residence and the breeding of mosquito larvae by surveying 299 households from different types of residences. The results are shown in <a href="#tbl-mos-larvae" class="quarto-xref">Table&nbsp;<span>20.1</span></a> . Please use an appropriate statistical method to analyze this data.</p>
</div>
<div>
<p><a href="datasets/ex20-04.csv" download="ex20-04.csv" class="btn btn-success"> <i class="bi bi-database-fill-down"></i> Download data </a></p>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>tb <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"datasets/ex20-04.csv"</span>, <span class="at">col_types =</span> <span class="fu">list</span>(<span class="at">resType =</span> <span class="fu">col_factor</span>())) <span class="sc">|&gt;</span> </span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">resType =</span> <span class="fu">factor</span>(resType, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"rural"</span>, <span class="st">"barriada"</span>, <span class="st">"urban"</span>))</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div id="tbl-mos-larvae" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-mos-larvae-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;20.1: Breeding situation of mosquito larvae in different resident areas
</figcaption>
<div aria-describedby="tbl-mos-larvae-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="do-not-create-environment cell caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 16%">
<col style="width: 20%">
<col style="width: 24%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">nContainer</th>
<th style="text-align: center;">nFamily.rural</th>
<th style="text-align: center;">nFamily.barriada</th>
<th style="text-align: center;">nFamily.urban</th>
<th style="text-align: center;">nFamily.total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">0</td>
<td style="text-align: center;">136</td>
<td style="text-align: center;">38</td>
<td style="text-align: center;">67</td>
<td style="text-align: center;">241</td>
</tr>
<tr class="even">
<td style="text-align: center;">1</td>
<td style="text-align: center;">23</td>
<td style="text-align: center;">8</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">36</td>
</tr>
<tr class="odd">
<td style="text-align: center;">2</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">12</td>
</tr>
<tr class="even">
<td style="text-align: center;">3</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">5</td>
</tr>
<tr class="odd">
<td style="text-align: center;">4</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">2</td>
</tr>
<tr class="even">
<td style="text-align: center;">5</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="odd">
<td style="text-align: center;">6</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="even">
<td style="text-align: center;">11</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="odd">
<td style="text-align: center;">Total</td>
<td style="text-align: center;">179</td>
<td style="text-align: center;">48</td>
<td style="text-align: center;">72</td>
<td style="text-align: center;">299</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
<p>You can fit a negative binomial regression using the <code>glm.nb()</code> function from the <code>MASS</code> package:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>neg.bin_fit <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(nContainer <span class="sc">~</span> resType, <span class="at">data =</span> tb, <span class="at">weights =</span> nFamily)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>poisson_fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(nContainer <span class="sc">~</span> resType, <span class="at">family =</span> poisson, <span class="at">data =</span> tb, <span class="at">weights =</span> nFamily)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(neg.bin_fit, poisson_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt;             df      AIC
#&gt; neg.bin_fit  4 426.2283
#&gt; poisson_fit  3 505.9154</code></pre>
</div>
</div>
<p>AIC is a measure used for model selection. A lower AIC value indicates a better-fitting model while penalizing for model complexity. In this case, the AIC for <code>neg.bin_fit</code> is 426.2283, which is lower than the AIC for poisson_fit, which is 505.9154. Since a lower AIC value is preferred, the negative binomial regression model fits the data better than the Poisson regression model.</p>
<p>The regression coefficients is:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(neg.bin_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt;     (Intercept) resTypebarriada    resTypeurban 
#&gt;      -0.7100490      -0.6762454      -1.9571792</code></pre>
</div>
</div>
<p>The regression equation is:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>equatiomatic<span class="sc">::</span><span class="fu">extract_eq</span>(neg.bin_fit, <span class="at">use_coefs =</span> T, <span class="at">coef_digits =</span> <span class="dv">4</span>,<span class="at">font_size =</span> <span class="st">"small"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><span class="math display">
\small
\begin{aligned}
\log ({ \widehat{E( \operatorname{nContainer} )} })  &amp;= -0.71 - 0.6762(\operatorname{resType}_{\operatorname{barriada}}) - 1.9572(\operatorname{resType}_{\operatorname{urban}})
\end{aligned}
</span></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(neg.bin_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#&gt; 
#&gt; Call:
#&gt; MASS::glm.nb(formula = nContainer ~ resType, data = tb, weights = nFamily, 
#&gt;     init.theta = 0.3002652205, link = log)
#&gt; 
#&gt; Coefficients:
#&gt;                 Estimate Std. Error z value Pr(&gt;|z|)    
#&gt; (Intercept)      -0.7100     0.1731  -4.102  4.1e-05 ***
#&gt; resTypebarriada  -0.6762     0.4274  -1.582 0.113612    
#&gt; resTypeurban     -1.9572     0.5256  -3.724 0.000196 ***
#&gt; ---
#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#&gt; 
#&gt; (Dispersion parameter for Negative Binomial(0.3003) family taken to be 1)
#&gt; 
#&gt;     Null deviance: 174.95  on 12  degrees of freedom
#&gt; Residual deviance: 156.37  on 10  degrees of freedom
#&gt; AIC: 426.23
#&gt; 
#&gt; Number of Fisher Scoring iterations: 1
#&gt; 
#&gt; 
#&gt;               Theta:  0.3003 
#&gt;           Std. Err.:  0.0764 
#&gt; 
#&gt;  2 x log-likelihood:  -418.2280</code></pre>
</div>
</div>
<p>Here is the interpretation of the results from the negative binomial regression output:</p>
<ol type="1">
<li><strong>Coefficients</strong>
<ul>
<li>Intercept: -0.7100, which corresponds to the log count of nContainer for the baseline group. p-value: 4.1e-05 (very small), which means the intercept is highly significant.</li>
<li>resTypebarriada (Effect of “barriada” residences compared to the baseline): -0.6762, indicating that the log count of containers is lower in barriada residences compared to the baseline by about 0.68. p-value = 0.113612, suggesting that the effect is not statistically significant.</li>
<li>resTypeurban (Effect of “urban” residences compared to the baseline): -1.9572, indicating that the log count of containers is significantly lower in urban residences compared to the baseline by about 1.96. p-value = 0.000196, suggesting this effect is statistically significant.</li>
</ul></li>
<li><strong>Model fit</strong>
<ul>
<li>Null deviance: 174.95 on 12 degrees of freedom, indicating how well the null model (no predictors) fits the data.</li>
<li>Residual deviance: 156.37 on 10 degrees of freedom, indicating how well the model with predictors fits the data. The reduction in deviance suggests that the predictors improve the model fit.</li>
<li>AIC: 426.23, a measure of model quality. Lower values suggest a better model fit, but should only be compared with other models fitted to the same data.</li>
</ul></li>
<li><strong>Dispersion parameter</strong>
<ul>
<li>Theta: 0.3003, with a standard error of 0.0764. Theta is the overdispersion parameter in the negative binomial model, and this small value indicates overdispersion, meaning the variance is larger than the mean, which justifies the use of negative binomial over Poisson regression.</li>
</ul></li>
</ol>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Conclusion
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>The baseline category (likely a reference category like rural) has a significant effect on the number of containers.</li>
<li>Urban residences have significantly fewer containers compared to the baseline category.</li>
<li>Barriada residences do not significantly differ from the baseline in terms of the number of containers.</li>
<li>The significant overdispersion in the data justifies using the negative binomial model over the Poisson model.</li>
</ul>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/qbgaoo\.github\.io\/r4ms");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      const annoteTargets = window.document.querySelectorAll('.code-annotation-anchor');
      for (let i=0; i<annoteTargets.length; i++) {
        const annoteTarget = annoteTargets[i];
        const targetCell = annoteTarget.getAttribute("data-target-cell");
        const targetAnnotation = annoteTarget.getAttribute("data-target-annotation");
        const contentFn = () => {
          const content = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          if (content) {
            const tipContent = content.cloneNode(true);
            tipContent.classList.add("code-annotation-tip-content");
            return tipContent.outerHTML;
          }
        }
        const config = {
          allowHTML: true,
          content: contentFn,
          onShow: (instance) => {
            selectCodeLines(instance.reference);
            instance.reference.classList.add('code-annotation-active');
            window.tippy.hideAll();
          },
          onHide: (instance) => {
            unselectCodeLines();
            instance.reference.classList.remove('code-annotation-active');
          },
          maxWidth: 300,
          delay: [50, 0],
          duration: [200, 0],
          offset: [5, 10],
          arrow: true,
          appendTo: function(el) {
            return el.parentElement.parentElement.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto',
          placement: 'right',
          popperOptions: {
            modifiers: [
            {
              name: 'flip',
              options: {
                flipVariations: false, // true by default
                allowedAutoPlacements: ['right'],
                fallbackPlacements: ['right', 'top', 'top-start', 'top-end', 'bottom', 'bottom-start', 'bottom-end', 'left'],
              },
            },
            {
              name: 'preventOverflow',
              options: {
                mainAxis: false,
                altAxis: false
              }
            }
            ]        
          }      
        };
        window.tippy(annoteTarget, config); 
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./logistic-regression.html" class="pagination-link" aria-label="Logistic regression analysis">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Logistic regression analysis</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./survival-analysis.html" class="pagination-link" aria-label="Survival analysis">
        <span class="nav-page-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Survival analysis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://qbgaoo.github.io/r4ms/edit/main/generalized-linear-model.qmd" class="toc-action"><i class="bi bi-git"></i>Edit this page</a></li><li><a href="https://qbgaoo.github.io/r4ms/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div></div></footer><script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>